fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/TemporaryOCFLRepositoryHelper.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.persistence.ocfl;
+
+import static org.fcrepo.persistence.ocfl.impl.OCFLConstants.OCFL_STAGING_PROPERTY;
+import static org.fcrepo.persistence.ocfl.impl.OCFLConstants.OCFL_STORAGE_ROOT_PROPERTY;
+import static org.fcrepo.persistence.ocfl.impl.OCFLConstants.OCFL_WORK_PROPERTY;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.apache.commons.io.FileUtils;
+
+/**
+ * Utility which sets up a temporary base directory for OCFL, establishes
+ * system properties to make use of it, and cleans up on shutdown.
+ *
+ * @author bbpennel
+ */
+public class TemporaryOCFLRepositoryHelper {
```

Comment:
This class is only used in tests. It would be ideal if we could move it into `src/text/java/...` ...which might require having the modules that use it in tests to include a "test-jar", such as:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-commons/pom.xml#L281-L303
and
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-integration-rdf/pom.xml#L62-L68

Reply:
I had considered doing that, but wasn't sure I wanted to add fcrepo-persistence-ocfl as a test dependency everywhere. Guess I should get over that though, I'll add it.

Reply:
Would it be possible to set the system-properties in the surefire/pom.xml configurations?
```
        System.setProperty(OCFL_STAGING_PROPERTY, ocflStagingDir.toString());
        System.setProperty(OCFL_STORAGE_ROOT_PROPERTY, ocflStorageRootDir.toString());
        System.setProperty(OCFL_WORK_PROPERTY, ocflWorkDir.toString());
```

..or some such thing?

Reply:
Hmm, it is possible, but it doesn't seem to pick up the properties when i run the tests in my IDE, only when I run them via maven. Sounds like eclipse may simply not support it:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=388683
Not sure if intellij is more observant. I might be inclined to stick with the utility since I almost always run tests via my IDE, but what do you think? I can remove the cleanup on shutdown to allow for inspection.

Reply:
If you'd like to test, this branch has surefire properties instead of the util:
https://github.com/fcrepo4/fcrepo4/tree/ocfl-temp
If you run any tests in FedoraLdpIT it will log the OCFL paths near the beginning of the run.

Reply:
An alternative is to tell people to setup debug configs in their IDEs with the system properties, I think that would work with relative paths. Any preference?